include/oneapi/mkl/rng/distributions.hpp

```diff
@@ -61,17 +61,24 @@ template <typename Type = float, typename Method = uniform_method::by_default>
 class uniform {
 public:
     static_assert(std::is_same<Method, uniform_method::standard>::value ||
-                      (std::is_same<Method, uniform_method::accurate>::value &&
-                       !std::is_same<Type, std::int32_t>::value),
+                      std::is_same<Method, uniform_method::accurate>::value,
                   "rng uniform distribution method is incorrect");
 
-    static_assert(std::is_same<Type, float>::value || std::is_same<Type, double>::value,
+    static_assert(std::is_same<Type, float>::value || std::is_same<Type, double>::value ||
+                      std::is_same<Type, std::int32_t>::value ||
+                      std::is_same<Type, std::uint32_t>::value,
                   "rng uniform distribution type is not supported");
 
     using method_type = Method;
     using result_type = Type;
 
-    uniform() : uniform(static_cast<Type>(0.0f), static_cast<Type>(1.0f)) {}
+    uniform()
+            : uniform(static_cast<Type>(0.0f),
+                      std::is_integral<Type>::value
```

Comment:
Do we need to add similar `if` for 64 bit integers here similar to device/distributions/hpp ?

Reply:
Reverting changes for host API. Let's concentrate on device API only